### DISCUSSION TITLE
Facing ⚠ Multiple root labels during data debug

### DISCUSSION THREAD

**Discussion by tohidmonjazeb** [explosion/spaCy#9747] [Help: Other Questions] [answered]
## How to reproduce the behaviour

python -m spacy debug data ./config.cfg

## Your Environment

* Operating System: windows 10
* Python Version Used: 3.8.5
* spaCy Version Used: 3.1.3
* Environment Information:
* """
Language: fa
Training pipeline: tok2vec, morphologizer, parser
26196 training docs
1456 evaluation docs
⚠ 2 training examples also in evaluation data

============================== Vocab & Vectors ==============================
ℹ 452495 total word(s) in the data (32017 unique)
ℹ No word vectors present in the package

========================= Morphologizer (POS+Morph) =========================
ℹ 15 label(s) in train data

============================= Dependency Parsing =============================
ℹ Found 452495 sentence(s) with an average length of 1.0 words.
ℹ 35 label(s) in train data
ℹ 35 label(s) in projectivized train data
⚠ Low number of examples for label 'goeswith' (1)
⚠ Low number of examples for label 'iobj' (6)
⚠ Multiple root labels (case, obl, det, nmod, amod, nsubj, cc, conj, punct,
advmod, appos, compound:lvc, root, flat:name, obj, xcomp, ccomp, obl:arg,
nummod, mark, acl, cop, aux:pass, nsubj:pass, aux, dep, advcl, parataxis,
flat:num, csubj, goeswith, fixed, vocative, compound, iobj) found in training
data. spaCy's parser uses a single root label ROOT so this distinction will not
be available.

================================== Summary ==================================
✔ 2 checks passed
⚠ 5 warnings

**Could anybody please tell me why I'm facing ⚠ Multiple root labels warning, and how to fix that ?**

**Comment by adrianeboyd** [comment, marked as answer]
Hi, it looks like something has gone wrong with your data conversion. It looks like every single word thinks it's the head of its own one-word sentence:

```
Found 452495 sentence(s) with an average length of 1.0 words.
```

My guess would be that the `token.head` values weren't set correctly in your conversion?

**Reply by tohidmonjazeb** [reply]
Actually I didn't set the heads cause I thought they aren't necessary , Are they?
Can setting them solve Multiple root labels warning ?

**Reply by adrianeboyd** [reply]
Yes, heads are necessary for dependency parses. I can recommend an introductory chapter from Speech and Language Processing by Jurafsky and Martin on this topic: https://web.stanford.edu/~jurafsky/slp3/14.pdf

**Reply by tohidmonjazeb** [reply]
I’m so thankful for your guidance and I'm going to try this ; Could you please keep the discussion open till I tell you the result?
